ui/components/app/confirmation/confirm-footer/confirm-footer.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import classnames from 'classnames';
+import {
+  Box,
+  Button,
+  ButtonSize,
+  ButtonVariant,
+} from '../../../component-library';
+import {
+  BlockSize,
+  Display,
+  FlexDirection,
+} from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { BoxProps } from '../../../component-library/box';
+import { ConfirmFooterProps } from './confirm-footer.types';
+
+const ConfirmFooter: React.FC<ConfirmFooterProps> = ({
+  className = '',
+  cancelText,
+  confirmText,
+  onCancel,
+  onConfirm,
+  confirmButtonProps,
+  cancelButtonProps,
+  ...props
+}) => {
+  const t = useI18nContext();
+  return (
+    <Box
+      className={classnames('confirm-footer', className)}
+      display={Display.Flex}
+      flexDirection={FlexDirection.Row}
+      gap={4}
+      padding={4}
+      width={BlockSize.Full}
+      {...(props as BoxProps<'div'>)}
+    >
+      <Button
+        variant={ButtonVariant.Secondary}
+        size={ButtonSize.Lg}
+        block
+        onClick={onCancel}
+        {...cancelButtonProps}
```

Comment:
This should resolve the TypeScript error by ensuring compatibility between button sizes and variants. It also allows for size overrides via `cancelButtonProps`.

```suggestion
        variant={ButtonVariant.Secondary}
        block
        onClick={onCancel}
        {...cancelButtonProps} // Resolves TS error
        size={ButtonSize.Lg || cancelButtonProps?.size} // Allows size override via cancelButtonProps
```